packages/js-client/GUIDE.md

```diff
@@ -0,0 +1,448 @@
+# Step-by-step guide to build a JS client
+
+---
+
+## Introduction
+
+The main goal of this guide is to show how to build a JS client for your Aragon OSx plugin. This guide assumes that you have already created a plugin and that you have a basic understanding of how Aragon OS works.
+
+An Aragon OSx plugin is composed of four mainly components, not all of them are mandatory:
+
+- Contracts
+- JS Client (optional)
+- Subgraph (optional)
+- UI (optional)
+
+The JS client sits as a middle layer between the contracts and the UI. It is responsible for fetching data from the contracts and exposing it to the UI. It also handles the communication between the UI and the contracts.
+
+Using this template and the `@aragon/sdk-client-common` package, you will be able to build a JS client for your plugin in a few minutes.
+
+The `@aragon/sdk-client-common` package provides a set tools to build a JS client for your plugin, it mainly gives you helpers and battle tested logic for handling connections with external services such as:
+
+- Web3 providers
+- Subgraph
+- IPFS
+
+This package handles for you the connection (fail prevention, error handling, connection retry...) and basic functionality of these services, so you can focus on building your plugin.
+
+### Why a JS client?
+
+As mentioned before, the JS client is a totally optional step in the process of creating an Aragon OS plugin. You can definitely build a plugin without a JS client, but you will be missing some benefits.
+
+Aragon common packages will handle the connection to IPFS, Subgraph and blockchain providers, so you don't have to reinvent the wheel.
+
+Additionally, is giving some structure to your plugin, if you plan on doing some complex logic in your plugin, it is a good idea to have a JS client to handle it and leave the UI handling only the UI logic.
+
+## Getting started
+
+To get started with your client the first step is to create a new JS/TS project you can use any tool you want.
+
+Once you have your project set up you need to install the `@aragon/sdk-client-common` package.
+
+```bash
+npm install @aragon/sdk-client-common
+// or
+yarn add @aragon/sdk-client-common
+```
+
+Once you have the package installed you can start building your client.
+
+There are 3 main files that you will need to build your client:
```

Comment:
Isn't this the wrong approach/writing style? The idea of this template is that everything is provided and already works. 
This guide sounds like you need to do all of that before you can start. But the thing is, that we already created it for the user.